### DISCUSSION TITLE
view의 인스턴스들을 class private fields로 관리를 해줘야 하는 대상인가요?! class private fields로 관리를 꼭 해줘야한다면 이유가 어떤것일까요..?!

### DISCUSSION THREAD

**Discussion by kkojae91** [woowacourse-study/2022-code-review-study#29] [Lotto] [answered]
크루분들의 생각이 궁금하네요!!

```js
export default class LottoController {
  #MoneyInputView = new MoneyInputView(
    $(`.${SELECTOR.CLASS.LOTTO_MONEY_SECTION}`)
  );
  #LottoListView = new LottoListView(
    $(`.${SELECTOR.CLASS.LOTTO_LIST_SECTION}`)
  );
  #WinningNumberInputView = new WinningNumberInputView(
    $(`.${SELECTOR.CLASS.WINNING_NUMBER_SECTION}`)
  );
  #ResultModalView = new ResultModalView($(`.${SELECTOR.CLASS.MODAL}`));
  #LottosModel = new LottosModel();
  #WinningLottoCounter = new WinningLottoCounter();
}
``;
```

**Comment by lokba** [comment, marked as answer]
좋은 질문입니다. 꼬재ㅎㅎ

제 생각을 애기드리겠습니다.

class 객체를 왜 쓰는걸까요? 
저는 객체지향 프로그래밍을 하고싶어서가 이유 중에 하나라고 생각합니다.

객체 지향이 나오면?
많은 리뷰어분들이, 객체 지향의 관점에서 `외부에서 접근하지 않는것은 모두 은닉의 대상이 된다`라고 피드백을 남기십니다.

그리하여, 저도 썩 끌리지 않지만 관리 대상이라고 생각합니다.

**Comment by wonsss** [comment]
`객체지향`과 `주체지향` 개념을 구분해보면 답이 나올 것 같아요.
- 주체지향 : 내가 모든 것을 일일이 다 처리하겠다.
- 객체지향 : 남을 시켜서 처리하겠다.
	- 그런데 남이 그 일을 어떤 방법으로 처리할지는 몰라도 된다. 그저 남에게 시키기만 한다.
	- 예를 들어, 코끼리를 냉장고에 넣어야 하는 문제가 있을 때, 교수님은 그저 '조교에게 시켜라'고만 명령하겠죠. 이렇듯 남에게 시켜버리면 그만인 게 객체지향 설계 방법일 것 같아요. 그 조교가 일을 어떻게 처리할지 하나하나 간섭하고 신경쓰려면 뭐하러 조교에게 시킬까요. 교수님이 직접 하면 그만이죠. 따라서 교수님은 조교가 무슨 방법으로 일을 처리할지는 "몰라도" 됩니다. 이 개념에서 객체의 `은닉화`라는 개념이 나오는 것 아닐까요?

따라서 객체 혼자서만 알아도 되는 것들은 private 으로 관리하고, 다른 객체로부터 반드시 받아야 하는 메시지로 볼 수 있는 메서드만 한정하여 public으로 관리하면 될 것 같아요. 

다음 말이 객체지향을 잘 설명해주는 것 같아요. 
"각자의 역할에만 충실해라. 그러면 이 세상은 돌아간다."

**Comment by moonheekim0118** [comment]
예전에 리뷰에서 본 내용인데.. 주전자를 사용할 때 사용자가 주전자 내부의 원리를 알 필요 없는 것 과 마찬가지아닐까요?
단순히 view 인스턴스 뿐만 아니라 외부에서 직접적으로 조작이 필요하지 않은 부분은 은닉화 시켜주는게 맞다고 생각해요 :)
